pinot-controller/src/main/java/org/apache/pinot/controller/validation/RealtimeSegmentValidationManager.java

```diff
@@ -108,8 +112,32 @@ protected void processTable(String tableNameWithType, Context context) {
     if (context._runSegmentLevelValidation) {
       runSegmentLevelValidation(tableConfig, streamConfig);
     }
+
+    // Keeps the table paused/unpaused based on storage quota validation.
+    // Skips updating the pause state if table is paused by admin
+    PauseState pauseState = computePauseState(tableNameWithType);
+
     _llcRealtimeSegmentManager.ensureAllPartitionsConsuming(tableConfig, streamConfig,
-        context._recreateDeletedConsumingSegment, context._offsetCriteria);
+        context._recreateDeletedConsumingSegment || !pauseState.isPaused(), context._offsetCriteria);
```

Comment:
Please change this to  _not_ call `ensureAllPartitionsConsuming` if the table is paused (or storage is exceeeded) rather than overloading the paused boolean with recreation of deleted segment.